### PR TITLE
Devices Pagination: Coverage and fixes of devices search/filter/query

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Follow up to https://github.com/flowforge/flowforge/pull/2677 that adds full server side test coverage after issues on staging causes by postgres (cypress uses sqlite).

## Related Issue(s)

https://github.com/flowforge/flowforge/issues/2379

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/db/models/Device.js [hunk 1]
```diff
@@ -291,7 +297,7 @@ module.exports = {
 
                     const [rows, count] = await Promise.all([
                         this.findAll({
-                            where: buildPaginationSearchClause(pagination, where, ['Device.name', 'Device.type', 'Device.id'], {}, order),
+                            where: buildPaginationSearchClause(pagination, where, ['Device.name', 'Device.type'], {}, order),
```
Comment (Pezmc) [inline forge/db/models/Device.js:300]: Removed as there's no way to search via hashid _in_ SQL.

**Review by Steve-Mcl — DISMISSED** [review]

**Comment by Steve-Mcl** [comment]
@Pezmc test fail on PG:

```
  2) Team Devices API
       Team Devices
         Supports sorting the devices
           by instance name:
     AssertionError: expected Array [ 'instance-2', 'project1', 'project1', undefined ] to match Array [ undefined, 'instance-2', 'project1', 'project1' ]
    not matched properties: '0' ('instance-2'), '1' ('project1'), '3' (undefined)
    matched properties: '2'
      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value [as match] (node_modules/should/cjs/should.js:356:19)
      at Context.<anonymous> (test/unit/forge/routes/api/teamDevices_spec.js:205:74)
      at runMicrotasks (<anonymous>)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

File: test/unit/forge/routes/api/teamDevices_spec.js [hunk 2]
```diff
@@ -116,7 +115,199 @@ describe('Team Devices API', function () {
...
+                const resultNameAsc = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=name&order=asc`)
+                resultNameAsc.map((device) => device.name).should.match(['device 1', 'device 2', 'device 3', 'device 4'])
+            })
+
+            it('by instance->application name', async function () {
+                // Sort by application name ASC (default)
+                const resultName = await queryDevices(`/api/v1/teams/${TestObjects.ATeam.hashid}/devices?sort=application`)
+
+                // SQLite puts undefined includes first ASC, Postgres has it last...
+                const usingSQLite = resultName[0].application === undefined
```
Comment (Steve-Mcl) [inline test/unit/forge/routes/api/teamDevices_spec.js:193]: This looks a little brittle to me. Why not check the database type instead?  `getDialect()` perhaps?

**Review by Steve-Mcl — APPROVED** [review]
Approved (awaiting ui test completion)

**Pull request merged**